compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc

```diff
@@ -0,0 +1,69 @@
+#include <TargetConditionals.h>
+#if TARGET_OS_OSX || TARGET_OS_IPHONE
+#include <dispatch/dispatch.h>
+#include <sys/sysctl.h>
+
+static bool isKnownAndSupported(const char *name) {
+  int32_t val = 0;
+  size_t size = sizeof(val);
+  if (sysctlbyname(name, &val, &size, NULL, 0))
+    return false;
+  return val;
+}
+
+void __init_cpu_features_resolver(void) {
+  // On Darwin platforms, this may be called concurrently by multiple threads
+  // because the resolvers that use it are called lazily at runtime (unlike on
+  // ELF platforms, where IFuncs are resolved serially at load time).  This
+  // function's effect on __aarch64_cpu_features should be idempotent, but even
+  // so we need dispatch_once to resolve the race condition.  Dispatch is
+  // available through libSystem, which we need anyway for the sysctl, so this
+  // does not add a new dependency.
+
+  static dispatch_once_t onceToken = 0;
+  dispatch_once(&onceToken, ^{
+    // https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics
+    static struct {
+      const char *sysctl_name;
+      enum CPUFeatures feature;
+    } features[] = {
```

Comment:
Do you need support for AdvSIMD_HPFPCvt, FEAT_LSE2,  FEAT_SHA512 or for other feature in Function Multi Versioning? Specification is not final yet and open to proposals.

Reply:
Sure, I can add entries for those. Looks like I missed them in my manual intersection of the [sysctl docs](https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics) with the `FEAT_*` table.

> Specification is not final yet and open to proposals.

Any word on when FMV might come out of beta?

Reply:
Ohh, I see you're asking a different question than I thought. Yeah, those three would probably be useful. Unfortunately, looks like we're just about out of bits in `__aarch64_cpu_features.features`.

Reply:
Another thing that came up when discussing the frontend attributes for these: we may want to have user-selectable overrides for the priorities for (hypothetical) cases where the prescribed numbers in the table don't match the selection behavior we need at runtime. I don't have a concrete example in mind yet.

Reply:
> Looks like we're just about out of bits in `__aarch64_cpu_features.features`.
Next field features2 can be added to __aarch64_cpu_features with setting FEAT_EXT bit to 1 and changing initialization condition to if (__aarch64_cpu_features.features &  (1ULL << FEAT_INIT) )

Reply:
> Any word on when FMV might come out of beta?

There's no specific deadline right now, possibly within a year or half - we want to collect and address all feedback on FMV from partners and compilers communities.